ext/nnstreamer/tensor_decoder/tensordec-tensor_region.c

```diff
@@ -0,0 +1,770 @@
+
+  g_array_free (results, TRUE);
+
+  gst_memory_unmap (out_mem, &out_info);
+
+  /** converting to Flexible tensor since  
+   * info pad of tensor_crop has capability for flexible tensor stream 
+  */
+  init_meta(&meta, trData);
+  out_mem = gst_tensor_meta_info_append_header(&meta, out_mem);
+
+  if (need_output_alloc)
+    gst_buffer_append_memory (outbuf, out_mem);
+  else
+    gst_memory_unref (out_mem);
+
+  return GST_FLOW_OK;
+error_unmap:
+  gst_memory_unmap (out_mem, &out_info);
+error_free:
+  gst_memory_unref (out_mem);
+
+  return GST_FLOW_ERROR;
+}
+
+/**
+ * @brief set the max_detection
+ */
+static int
+_set_max_detection (tensor_region *data, guint max_detection, unsigned int limit)
+{
+  /**Check consistency with max_detection */
+  if (data->max_detections == 0)
+    data->max_detections = max_detection;
+  else
+    g_return_val_if_fail (max_detection == data->max_detections, FALSE);
+
+  if (data->max_detections > limit) {
+    GST_ERROR ("Incoming tensor has too large detection-max : %u", max_detection);
+    return FALSE;
+  }
+  return TRUE;
+}
+
+
+/**
+ * @brief tensordec-plugin's GstTensorDecoderDef callback
+ *
+ * [Mobilenet SSD Model]
+ * The first tensor is boxes. BOX_SIZE : 1 : #MaxDetection, ANY-TYPE
+ * The second tensor is labels. #MaxLabel : #MaxDetection, ANY-TYPE
+ * Both tensors are MANDATORY!
+ * If there are third or more tensors, such tensors will be ignored.
+ */
+
+static GstCaps *
+tr_getOutCaps (void **pdata, const GstTensorsConfig *config)
+{
+  tensor_region *data = *pdata;
+  GstCaps *caps;
+  char *str;
+  guint max_detection, max_label;
+  const uint32_t *dim1, *dim2;
+  int i;
+
+  /**Check if the first tensor is compatible */
+  dim1 = config->info.info[0].dimension;
+  g_return_val_if_fail (dim1[0] == BOX_SIZE, NULL);
+  g_return_val_if_fail (dim1[1] == 1, NULL);
+  max_detection = dim1[2];
+  g_return_val_if_fail (max_detection > 0, NULL);
+  for (i = 3; i < NNS_TENSOR_RANK_LIMIT; i++)
+    g_return_val_if_fail (dim1[i] == 1, NULL);
+
+  /**Check if the second tensor is compatible */
+  dim2 = config->info.info[1].dimension;
+  max_label = dim2[0];
+  g_return_val_if_fail (max_label <= data->labeldata.total_labels, NULL);
+  if (max_label < data->labeldata.total_labels)
+    GST_WARNING ("The given tensor (2nd) has max_label (first dimension: %u) smaller than the number of labels in labels file (%s: %u).",
+        max_label, data->label_path, data->labeldata.total_labels);
+  g_return_val_if_fail (max_detection == dim2[1], NULL);
+  for (i = 2; i < NNS_TENSOR_RANK_LIMIT; i++)
+    g_return_val_if_fail (dim2[i] == 1, NULL);
+
+  /**Check consistency with max_detection */
+  if (!_set_max_detection (data, max_detection, MOBILENET_SSD_DETECTION_MAX)) {
+    return NULL;
+  }
+  str = g_strdup_printf("other/tensor-flexible, width=%u, height=%u, type=(int)32, dimension=%d:%d", data->i_width, data->i_height, BOX_SIZE, data->num);

```

Comment:
width, height, type and information are written in header of flexible tensors.
```suggestion
  str = g_strdup_printf(other/tensors,format=flexible");

```

Reply:
I have updated the string value. Thank you for suggesting this change.